app/views/articles/tags/_meta.html.erb

```diff
@@ -3,12 +3,12 @@
      timeframe: params[:timeframe],
      content_for: true,
    ) %>
-<link rel="canonical" href="https://dev.to/t/<%= @tag %>" />
+<link rel="canonical" href="https://dev.to/t/<%= @tag %><%= "/page/#{@page}" if @page > 1 %>" />
```

Comment:
Note that we can still use our helper here:

```ruby
app_url([@tag, ("/#{@page}" if @page > 1)].join)
```

Here you can see it in action:

```ruby
@page = 1
app_url([@tag, ("/#{@page}" if @page > 1)].join)
#=> "http://localhost:3000/ruby"

@page = 2
helper.app_url([@tag, ("/#{@page}" if @page > 1)].join)
"http://localhost:3000/ruby/2"
```

I'd maybe turn that into a helper:

```ruby
def tag_page_url(tag, page)
  app_url([tag, ("/#{page}" if page > 1)].join)
end
```

which makes for pretty nice callsites:

```ruby
<link rel="canonical" href="<%= tag_page_url(@tag, @page) %>" />
```

Reply:
✅